modules/cudaarithm/src/cuda/math.cu

```diff
@@ -278,43 +278,26 @@ namespace
 {
     template<typename T, bool Signed = numeric_limits<T>::is_signed> struct PowOp : unary_function<T, T>
     {
-        float power;
+        T power;
```

Comment:
`T` can be `uint8_t`
`power` can be `0.5f`.

"power" changes should be reverted.